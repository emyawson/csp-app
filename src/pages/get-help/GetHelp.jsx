/* eslint-disable react/prop-types */
// Modules
import React, { useState, useEffect, Fragment } from 'react';
import { useSelector } from 'react-redux';
import { withRouter, useHistory } from 'react-router-dom';
import ActionList from '@ingka/action-list';
import Layout from '../../components/layout/Layout.jsx';
import BCrumb from '../../components/Bcrumb/BCrumb.jsx';
import Headline from '../../components/headline/Headline.jsx';
import StatusOrder from '../../components/statusOrder/StatusOrder.jsx';
import Topic from '../../components/topic/Topic.jsx';

import { selectPages, selectSelectedTopic, selectLoadingPages } from '../../store/selectors';
import { SET_SELECTED_CONTACT_ID } from '../../store/actions';
import { ROUTE_GET_SUPPORT } from '../../constants/routes';
import { store } from '../../store';

// styles
import '../home/Home.scss';
import LoadingWrapper from '../../components/loading/LoadingWrapper.jsx';
import { GTM_DATA } from '../../constants/GTMmarket';
import { gtmTopic, gtmActionsData } from '../../utils/gtm';

const HeadlineLoader = LoadingWrapper(Headline);
const StatusOrderLoader = LoadingWrapper(StatusOrder);
const TopicLoader = LoadingWrapper(Topic);

const GetHelp = () => {
  const loadingPages = useSelector(selectLoadingPages);
  const pages = useSelector(selectPages);
  // Recommended topics of the current selected topic

  const topicID = useSelector(selectSelectedTopic);
  const [selectedTopicDetails, setSelectedTopicDetails] = useState({});
  const [hasRecommendation, setHasRecommendation] = useState({});

  const history = useHistory();

  const topicClickHandler = (contactId, index) => {
    // gtm
    gtmTopic(contactId, gtmActionsData(GTM_DATA), index);
    // next page
    store.dispatch({ type: SET_SELECTED_CONTACT_ID, payload: { contactId } });

    history.push(ROUTE_GET_SUPPORT);
  };
  useEffect(() => {
    if (pages) {
      setSelectedTopicDetails(pages.topics.find(({ id }) => id === topicID));
    }

    if (selectedTopicDetails) {
      setHasRecommendation(selectedTopicDetails.recommendation);
    }
  }, [pages, selectedTopicDetails]);
  return (
    <Layout>
      <BCrumb />
      <HeadlineLoader
        isLoading={loadingPages}
        loadingMarginTop="60px"
        loadingWidth="505px"
        loadingHeight="106px"
        title={selectedTopicDetails.headline}
        description={selectedTopicDetails.body}
      />
      {hasRecommendation ? (
        <StatusOrderLoader
        loadingHeight="106px"
          isLoading={loadingPages}
          heading={hasRecommendation.headline}
          content={hasRecommendation.body}
          urlButton={hasRecommendation.callToActionUrl}
          label={hasRecommendation.callToActionText}
        />
      ) : null}
      {selectedTopicDetails.topics ? (
        <ActionList className="topics-container" data-cytest="subtopics-list">
          {selectedTopicDetails.topics.map((suggestedTopic, index) => (
            <Fragment key={index}>
              <TopicLoader
                loadingHeight="106px"
                isLoading={loadingPages}
                label={suggestedTopic.topicHeadline}
                caption={suggestedTopic.topicBody}
                clickHandler={() => topicClickHandler(suggestedTopic.id, index)}
              />
            </Fragment>
          ))}
        </ActionList>
      ) : (
        ''
      )}
    </Layout>
  );
};
export default withRouter(GetHelp);
