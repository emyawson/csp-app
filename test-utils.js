import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
