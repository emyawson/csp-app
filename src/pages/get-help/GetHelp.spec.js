import React from 'react';
// import { render, screen } from "@testing-library/react";
import GetHelp from './GetHelp.jsx';
// import { renderWithRouter } from "../test-utils";
import { renderWithRouter } from '../../../test-utils';

import pagesMock from '../../services/mock/pagesMock';

const defaultState = {
  router: {},
  loading: {},
  pages: pagesMock,
  selectedTopic: '1L8HKugsJOw3bgoVI4txrc',
  selectedContactPage: '1SuaI8C7L4mQBQLrgFwsGr',
  breadcrumbPathname: [],
  errors: {},
};

const renderComponent = () => renderWithRouter(<GetHelp/>, {}, defaultState);

test('renders <GetHelp/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();
});
