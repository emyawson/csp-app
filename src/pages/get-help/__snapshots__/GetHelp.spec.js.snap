// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <GetHelp/> 1`] = `
<div>
  <main
    class="page-container"
  >
    <div
      class="page-container__inner"
    >
      <div
        class="page-container__main"
      >
        <div
          class="breadcrumb"
        >
          <nav
            aria-label="Breadcrumb"
            class="breadcrumb__nav"
            data-cytest="breadcrumb"
            role="navigation"
          >
            <ol
              class="breadcrumb__list"
              itemprop="breadcrumb"
              itemscope=""
              itemtype="http://schema.org/BreadcrumbList"
            >
              <li
                class="breadcrumb__list-item"
                itemprop="itemListElement"
                itemscope=""
                itemtype="http://schema.org/ListItem"
              >
                 
                <a
                  class="breadcrumb__link link link--black"
                  href="/nl/nl"
                  itemprop="item"
                >
                  
                  <span
                    class="breadcrumb__item"
                    data-cytes="crumb"
                  >
                    IKEA
                  </span>
                </a>
                 
                <meta
                  content="1"
                  itemprop="position"
                />
              </li>
              <li
                class="breadcrumb__list-item"
                itemprop="itemListElement"
                itemscope=""
                itemtype="http://schema.org/ListItem"
              >
                 
                <a
                  class="breadcrumb__link link link--black"
                  href="/nl/nl/customer-service"
                  itemprop="item"
                >
                  
                  <span
                    class="breadcrumb__item"
                    data-cytes="crumb"
                  >
                    Klantenservice
                  </span>
                </a>
                 
                <meta
                  content="2"
                  itemprop="position"
                />
              </li>
              <li
                class="breadcrumb__list-item"
                itemprop="itemListElement"
                itemscope=""
                itemtype="http://schema.org/ListItem"
              >
                 
                <a
                  class="breadcrumb__link link link--black"
                  href="/nl/nl/customer-service/support"
                  itemprop="item"
                >
                  
                  <span
                    class="breadcrumb__item"
                    data-cytes="crumb"
                  >
                    We are here to help and you should be greatful
                  </span>
                </a>
                 
                <meta
                  content="3"
                  itemprop="position"
                />
              </li>
            </ol>
          </nav>
        </div>
        <div
          class="headline"
        >
          <h1
            class="headline__title"
            data-cytest="headline"
          >
            Get support for IKEA store information and About IKEA
          </h1>
          <p
            class="headline__description"
          >
            Choose a topic and we’ll find you the best way to get in touch.
          </p>
        </div>
        <div
          class="status-order"
        >
          <p
            class="label"
          >
            Aanbevolen
          </p>
          <h2
            class="status-order__heading"
          >
            If you are looking for spare parts you can order them online
          </h2>
          <div
            class="richtext-editorstyle status-order__description"
          >
            <p>
              PH: Missing a leg for your sofa? Need an extra hinge? If you are looking for spare parts big or small, we’ve got you covered. Order your spare part online.
            </p>
          </div>
          <button
            class="btn btn--primary status-order__btn"
            type="button"
          >
            <span
              class="btn__inner"
            >
              
              Order spare parts
            </span>
          </button>
        </div>
        <ul
          class="action-list action-list--regular topics-container"
          data-cytest="subtopics-list"
        >
          <li
            class="action-list-item"
          >
            <button
              class="action-list-item__action"
            >
              <span
                class="action-list-item__label-container"
              >
                <span
                  class="action-list-item__label"
                >
                  Stores information
                </span>
                <span
                  class="action-list-item__caption"
                >
                  Opening hours and location
                </span>
              </span>
              <span
                class="action-list-item__chevron"
              >
                <svg
                  aria-hidden="true"
                  class="svg-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    clip-rule="evenodd"
                    d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li
            class="action-list-item"
          >
            <button
              class="action-list-item__action"
            >
              <span
                class="action-list-item__label-container"
              >
                <span
                  class="action-list-item__label"
                >
                  In-store activities
                </span>
                <span
                  class="action-list-item__caption"
                >
                  Find out what types of events and workshops are happening at IKEA
                </span>
              </span>
              <span
                class="action-list-item__chevron"
              >
                <svg
                  aria-hidden="true"
                  class="svg-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    clip-rule="evenodd"
                    d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li
            class="action-list-item"
          >
            <button
              class="action-list-item__action"
            >
              <span
                class="action-list-item__label-container"
              >
                <span
                  class="action-list-item__label"
                >
                  Campaigns & Offers
                </span>
                <span
                  class="action-list-item__caption"
                >
                  Missing descriptive body text
                </span>
              </span>
              <span
                class="action-list-item__chevron"
              >
                <svg
                  aria-hidden="true"
                  class="svg-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    clip-rule="evenodd"
                    d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li
            class="action-list-item"
          >
            <button
              class="action-list-item__action"
            >
              <span
                class="action-list-item__label-container"
              >
                <span
                  class="action-list-item__label"
                >
                  Lost & Found
                </span>
                <span
                  class="action-list-item__caption"
                >
                  Missing descriptive body text
                </span>
              </span>
              <span
                class="action-list-item__chevron"
              >
                <svg
                  aria-hidden="true"
                  class="svg-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    clip-rule="evenodd"
                    d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li
            class="action-list-item"
          >
            <button
              class="action-list-item__action"
            >
              <span
                class="action-list-item__label-container"
              >
                <span
                  class="action-list-item__label"
                >
                  About IKEA
                </span>
                <span
                  class="action-list-item__caption"
                >
                  Missing descriptive body text
                </span>
              </span>
              <span
                class="action-list-item__chevron"
              >
                <svg
                  aria-hidden="true"
                  class="svg-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    clip-rule="evenodd"
                    d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li
            class="action-list-item"
          >
            <button
              class="action-list-item__action"
            >
              <span
                class="action-list-item__label-container"
              >
                <span
                  class="action-list-item__label"
                >
                  SMÅLAND
                </span>
                <span
                  class="action-list-item__caption"
                >
                  Play area for kids
                </span>
              </span>
              <span
                class="action-list-item__chevron"
              >
                <svg
                  aria-hidden="true"
                  class="svg-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    clip-rule="evenodd"
                    d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li
            class="action-list-item"
          >
            <button
              class="action-list-item__action"
            >
              <span
                class="action-list-item__label-container"
              >
                <span
                  class="action-list-item__label"
                >
                  IKEA Family
                </span>
                <span
                  class="action-list-item__caption"
                >
                  Missing descriptive body text
                </span>
              </span>
              <span
                class="action-list-item__chevron"
              >
                <svg
                  aria-hidden="true"
                  class="svg-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    clip-rule="evenodd"
                    d="M15.6 12l-5.785 5.786L8.4 16.372l4.371-4.371-4.37-4.372 1.414-1.414L15.6 12z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </main>
</div>
`;
